src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -26,17 +26,10 @@ public class QnAService {
     @Transactional
     public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
         Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
+        question.checkOwner(question, loginUser);
+        question.checkAnswerOwner(loginUser);
```

Comment:
서비스 계층에서 직접 삭제가 가능한지 체크하는건 책임의 영역이 도메인이 아닌 서비스가 되는 것 같아요. 
그리고 두 함수는 검증 실패시 예외 던지는 행위가 예고되어있는데, 이를 아는데 대처하지 않고 외부로 그냥 전파하는건 적절하지 않은 것 같습니다. 

```suggestion
        List<DeleteHistory> deleteHistories = question.delete(loginUser);
        deleteHistoryService.saveAll(deleteHistories);
```

두 줄정도로도 충분히 표현 가능하지 않나요?